tests/src/test/scala/common/rest/WskRest.scala

```diff
@@ -0,0 +1,1437 @@
+   * if the code is anything but DONTCARE_EXIT, assert the code is as expected
+   */
+  override def delete(basepathOrApiName: String,
+                      relpath: Option[String] = None,
+                      operation: Option[String] = None,
+                      expectedExitCode: Int = SUCCESS_EXIT,
+                      cliCfgFile: Option[String] = None)(implicit wp: WskProps): RestResult = {
+    val parms = Map[String, JsValue]() ++ { Map("__ow_user" -> wp.namespace.toJson) } ++ {
+      Map("basepath" -> basepathOrApiName.toJson)
+    } ++ {
+      relpath map { r =>
+        Map("relpath" -> r.toJson)
+      } getOrElse Map[String, JsValue]()
+    } ++ {
+      operation map { o =>
+        Map("operation" -> o.toJson)
+      } getOrElse Map[String, JsValue]()
+    } ++ {
+      Map("accesstoken" -> wp.authKey.toJson)
+    } ++ {
+      Map("spaceguid" -> wp.authKey.split(":")(0).toJson)
+    }
+
+    val rr = invokeAction(
+      name = "apimgmt/deleteApi",
+      parameters = parms,
+      blocking = true,
+      result = true,
+      expectedExitCode = expectedExitCode)(wp)
+    return rr
+  }
+
+  def getApi(basepathOrApiName: String, params: Map[String, String] = null, expectedExitCode: Int = OK.intValue)(
+    implicit wp: WskProps): RestResult = {
+    val whiskUrl = Uri(s"http://${WhiskProperties.getBaseControllerHost()}:9001")
+    val path = Path(s"/api/${wp.authKey.split(":")(0)}$basepathOrApiName/path")
+    val resp = requestEntity(GET, path, params, whiskUrl = whiskUrl)
+    val result = new RestResult(resp.status, getRespData(resp))
+    result
+  }
+}
+
+class RunWskRestCmd() extends FlatSpec with RunWskCmd with Matchers with ScalaFutures with WskActorSystem {
+
+  implicit val config = PatienceConfig(10 seconds, 0 milliseconds)
+  implicit val materializer = ActorMaterializer()
+  val whiskRestUrl = Uri(s"http://${WhiskProperties.getBaseControllerAddress()}")
+  val basePath = Path("/api/v1")
+
+  def validateStatusCode(expectedExitCode: Int, statusCode: Int) = {
+    if ((expectedExitCode != DONTCARE_EXIT) && (expectedExitCode != ANY_ERROR_EXIT))
+      if (statusCode != expectedExitCode)
+        statusCode shouldBe expectedExitCode
+  }
+
+  def getNamePath(noun: String, name: String)(implicit wp: WskProps): Path = {
+    return Path(s"$basePath/namespaces/${wp.namespace}/$noun/$name")
+  }
+
+  def getExt(filePath: String)(implicit wp: WskProps) = {
+    val sep = "."
+    if (filePath.contains(sep)) filePath.substring(filePath.lastIndexOf(sep), filePath.length())
+    else null
+  }
+
+  def request(method: HttpMethod, uri: Uri, body: String = "", creds: BasicHttpCredentials): Future[HttpResponse] = {
+    val entity = HttpEntity(ContentTypes.`application/json`, body)
+    val request = HttpRequest(method, uri, List(Authorization(creds)), entity = entity)
+    Http().singleRequest(request)
+  }
+
+  def requestEntity(method: HttpMethod,
```

Comment:
Nice. I think the goal is to avoid using null if possible. Can you use `Some` and `None` instead?

Reply:
Done